### ISSUE TITLE
忠实老用户提两点细节改进建议,干货.很基本的功能,但对提升用户体验作用很大!

### ISSUE THREAD

**Issue by superDMS** [yingDev/WGestures2-mac-bugs#161]
1.添加重复手势无任何提示冲突, 以至于手势过多有时候重复添加后都不知道.  可加上无法添加重复手势的冲突提示, 确保唯一性. 
![image](https://user-images.githubusercontent.com/57837939/156083180-da91062e-7d76-4d1e-94a5-27a2604a5bf6.png)
![image](https://user-images.githubusercontent.com/57837939/155668647-f04cec5f-b4f4-49ab-aed6-75bbb9600a37.png)

2.现版本的手势提示仅能提示当前所能触发的手势, 辅助性比较弱!
应当显示当前手势命令分支下的所有手势具体触发命令.   
这样更方便记忆,特别实用. 
 有时候要用到一些不常用到的手势,经常突然忘记是什么了..有了这个分支脑图辅助性大大增加!以后可以大胆设置更多的手势, 只需记忆第一个命令, 即可呼出整个分支.  
效果如下:
![image](https://user-images.githubusercontent.com/57837939/155670970-788f2464-c075-4d89-a00d-5175894a8a95.png)
![image](https://user-images.githubusercontent.com/57837939/155674663-9683f513-992e-44f5-85bb-e1f05f868ee2.png)
![image](https://user-images.githubusercontent.com/57837939/155669094-9e696646-4640-420e-a1d4-3d1cbfe70808.png)
(如担心显示过多, 可做限制, 比如至多显示分支下20个手势,或可自定义显示数量, 不介意的话铺满全屏都无所谓,  显示形式可借鉴也可自定义)

这两点应该都非常简单! 不是特别刁钻的功能 !   更完善=更强大!

**Comment by superDMS** [comment]
dddddddd

**Comment by vr2009** [comment]
> 1.添加重复手势无任何提示冲突, 以至于手势过多有时候重复添加后都不知道. 可加上无法添加重复手势的冲突提示, 确保唯一性. ![image](https://user-images.githubusercontent.com/57837939/156083180-da91062e-7d76-4d1e-94a5-27a2604a5bf6.png) ![image](https://user-images.githubusercontent.com/57837939/155668647-f04cec5f-b4f4-49ab-aed6-75bbb9600a37.png)
> 


你用的这些手势真是大开眼界，都赶上格斗游戏了，确实要很大的脑力才能记住这些操作，一般人来说手势都是直上直下的，可能用不了这么复杂的手势

**Comment by vr2009** [comment]
我倒觉得作者能在触发菜单和屏幕边缘上下功夫，体验会提升更高层次。
比如菜单里的单个命令可以用短链接形式调用，这样菜单命令就能相互调用，编辑菜单就更简单了。另外菜单可以绑定快捷键就更好了，虽然WG2是鼠标手势软件，给内部命令绑定快捷键的做法也未尝不可。
屏幕边缘目前来是整个边缘触发相同的功能，现在的屏幕有长很高，很多区域是浪费的，把屏幕边缘划分成左中右，上中下，比如屏幕 底部边缘左边【右键】触发任务管理器，边缘中间【右键】触发文件管理器，边缘右边【右键】触发回到桌面，这样一来一个键在屏幕边缘能实现三个功能，用户记忆成本就低很多。
我的使用习惯是屏幕右侧边缘上半部分【滚轮上下】调节亮度，下半部分【滚轮上下】调节音量大小，下半部分【右键】静音。用很少记忆成本实现更多的功能。

**Comment by superDMS** [comment]
> > 1.添加重复手势无任何提示冲突, 以至于手势过多有时候重复添加后都不知道. 可加上无法添加重复手势的冲突提示, 确保唯一性. ![image](https://user-images.githubusercontent.com/57837939/156083180-da91062e-7d76-4d1e-94a5-27a2604a5bf6.png) ![image](https://user-images.githubusercontent.com/57837939/155668647-f04cec5f-b4f4-49ab-aed6-75bbb9600a37.png)
> 
> 你用的这些手势真是大开眼界，都赶上格斗游戏了，确实要很大的脑力才能记住这些操作，一般人来说手势都是直上直下的，可能用不了这么复杂的手势

是的  手势多到一定程度  相似的手势就容易搞混   命令太多了就会想各种其他分支.....   所以才想着让老大增加分支辅助提示

**Comment by JohnKhanJr** [comment]
相似手势的区分，我的作法是用字母键区分，比如默认↑↓◑是搜索，由此衍生出：↑↓◑B（百度搜索）、↑↓◑G（必应搜索）、↑↓◑Z（知乎搜索）……，很好记，缺点是不太适合单手党。

如果手势菜单能使用快捷键响应也可以利用起来，比如菜单的呼出手势为→◑◐，其每一项都能用鼠标上、下滚轮或按键激活执行，这样子，也可以避免大量无规则记忆的麻烦

**Comment by howboion** [comment]
> 1.添加重复手势无任何提示冲突, 以至于手势过多有时候重复添加后都不知道. 可加上无法添加重复手势的冲突提示, 确保唯一性. ![image](https://user-images.githubusercontent.com/57837939/156083180-da91062e-7d76-4d1e-94a5-27a2604a5bf6.png) ![image](https://user-images.githubusercontent.com/57837939/155668647-f04cec5f-b4f4-49ab-aed6-75bbb9600a37.png)
> 
> 2.现版本的手势提示仅能提示当前所能触发的手势, 辅助性比较弱! 应当显示当前手势命令分支下的所有手势具体触发命令. 这样更方便记忆,特别实用. 有时候要用到一些不常用到的手势,经常突然忘记是什么了..有了这个分支脑图辅助性大大增加!以后可以大胆设置更多的手势, 只需记忆第一个命令, 即可呼出整个分支. 效果如下: ![image](https://user-images.githubusercontent.com/57837939/155670970-788f2464-c075-4d89-a00d-5175894a8a95.png) ![image](https://user-images.githubusercontent.com/57837939/155674663-9683f513-992e-44f5-85bb-e1f05f868ee2.png) ![image](https://user-images.githubusercontent.com/57837939/155669094-9e696646-4640-420e-a1d4-3d1cbfe70808.png) (如担心显示过多, 可做限制, 比如至多显示分支下20个手势,或可自定义显示数量, 不介意的话铺满全屏都无所谓, 显示形式可借鉴也可自定义)
> 
> 这两点应该都非常简单! 不是特别刁钻的功能 ! 更完善=更强大!

我觉得第二点非常不错

**Comment by superDMS** [comment]
> > 1.添加重复手势无任何提示冲突, 以至于手势过多有时候重复添加后都不知道. 可加上无法添加重复手势的冲突提示, 确保唯一性. ![image](https://user-images.githubusercontent.com/57837939/156083180-da91062e-7d76-4d1e-94a5-27a2604a5bf6.png) ![image](https://user-images.githubusercontent.com/57837939/155668647-f04cec5f-b4f4-49ab-aed6-75bbb9600a37.png)
> > 2.现版本的手势提示仅能提示当前所能触发的手势, 辅助性比较弱! 应当显示当前手势命令分支下的所有手势具体触发命令. 这样更方便记忆,特别实用. 有时候要用到一些不常用到的手势,经常突然忘记是什么了..有了这个分支脑图辅助性大大增加!以后可以大胆设置更多的手势, 只需记忆第一个命令, 即可呼出整个分支. 效果如下: ![image](https://user-images.githubusercontent.com/57837939/155670970-788f2464-c075-4d89-a00d-5175894a8a95.png) ![image](https://user-images.githubusercontent.com/57837939/155674663-9683f513-992e-44f5-85bb-e1f05f868ee2.png) ![image](https://user-images.githubusercontent.com/57837939/155669094-9e696646-4640-420e-a1d4-3d1cbfe70808.png) (如担心显示过多, 可做限制, 比如至多显示分支下20个手势,或可自定义显示数量, 不介意的话铺满全屏都无所谓, 显示形式可借鉴也可自定义)
> > 这两点应该都非常简单! 不是特别刁钻的功能 ! 更完善=更强大!
> 
> 我觉得第二点非常不错

顶起来!

**Comment by jabobian** [comment]
> 

>菜单里的单个命令可以用短链接形式调用，这样菜单命令就能相互调用，编辑菜单就更简单了。另外菜单可以绑定快捷键就更>好了，虽然WG2是鼠标手势软件，给内部命令绑定快捷键的做法也未尝不可

全局url协议和短链接对于软件核心功能来说是正解。有了全局URL，外部程序可以方便的调用WG菜单（不要求鼠标手势触发，因为触发的时机可以取决于更多上下文）。短链接的话也是为了方便分组之间调出特定共用菜单等。此外，也方便了多个手势调出同一个自定义菜单，一个记不住，多一个可选的，不容易忘记。

其它功能绝大部分可以用菜单项来替代，比如一个基本手势弹出一个菜单（包含两列，第一列是扩展手势，第二列是文字说明）。然后每个扩展手势菜单项对应一个字母，应该已经够大部分使用。

**Comment by homjie123** [comment]
臣附议，顶，经常会遇到因重复手势没有提示而错误的情况
